### PR TITLE
Sweet Home 3D

### PR DESCRIPTION
**Pull request by hadess**
From http://www.sweethome3d.com/

### CODE REVIEW

**Comment by hadess** [comment]
bot, build com.sweethome3d.Desktop

File: com.sweethome3d.Desktop.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "sdk" : "org.freedesktop.Sdk",
+    "command" : "SweetHome3D",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--device=dri",
+        "--filesystem=home",
+         "--env=PATH=/usr/bin:/app/bin:/app/jdk/bin",
+         "--env=JAVA_HOME=/app/jdk"
```
Comment (barthalion) [inline com.sweethome3d.Desktop.json]: Just to make sure, does the application ship its own JDK copy?

File: com.sweethome3d.Desktop.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.sweethome3d.Desktop.appdata.xml]: Ideally it should be contributed upstream.

File: com.sweethome3d.Desktop.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            "sources" : [
+                {
+                    "type": "archive",
+                    "only-arches": ["x86_64"],
+                    "sha256": "5c2090b16c32e52637e87a48197a13d30a0322940949431bbb898b7b082cfca6",
+                    "url": "https://downloads.sourceforge.net/project/sweethome3d/SweetHome3D/SweetHome3D-6.2/SweetHome3D-6.2-linux-x64.tgz"
+                },
+                {
+                    "type": "archive",
+                    "only-arches": ["i386"],
```
Comment (barthalion) [inline com.sweethome3d.Desktop.json]: 19.08 dropped support for i386.

File: com.sweethome3d.Desktop.json (hunk 1 above)
Comment (hadess) [reply com.sweethome3d.Desktop.json]: It does. I should probably remove this, as this is already handled in the app's startup scripts.

File: com.sweethome3d.Desktop.appdata.xml (hunk 2 above)
Comment (hadess) [reply com.sweethome3d.Desktop.appdata.xml]: I'd rather have it reviewed first. As it is, there isn't even a .desktop file, so I'd rather propose everything upstream once I've got something to show for it.

**Comment by LeandroStanger** [comment]
bot, build com.sweethome3d.Desktop

File: com.sweethome3d.Desktop.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            "name": "sweethome3d",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir /app/lib/",
+                "cp -r ./ /app/lib/sweethome3d/",
+                "mkdir -p /app/bin/",
+                "ln -s /app/lib/sweethome3d/SweetHome3D /app/bin/",
+                "install -Dm644 SweetHome3DIcon.png /app/share/icons/com.sweethome3d.Desktop.png",
+                "install -Dm644 com.sweethome3d.Desktop.desktop /app/share/applications/com.sweethome3d.Desktop.desktop",
+                "install -Dm644 com.sweethome3d.Desktop.appdata.xml /app/share/appdata/com.sweethome3d.Desktop.appdata.xml",
```
Comment (bilelmoussaoui) [inline com.sweethome3d.Desktop.json]:
```suggestion
                "install -Dm644 com.sweethome3d.Desktop.metainfo.xml /app/share/metainfo/com.sweethome3d.Desktop.metainfo.xml",
```
The .appdata should be renamed to .metainfo

File: com.sweethome3d.Desktop.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+<!-- Copyright 2019 Bastien Nocera <hadess@hadess.net> -->
+<component type="desktop">
```
Comment (bilelmoussaoui) [inline com.sweethome3d.Desktop.appdata.xml]:
```suggestion
<component type="desktop-application">
```

File: com.sweethome3d.Desktop.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  <id>com.sweethome3d.Desktop</id>
```
Comment (bilelmoussaoui) [inline com.sweethome3d.Desktop.appdata.xml]: I think there were some issues when an app used .Desktop on their app-id but I don't remember if that was fixed. @TingPing do you know what was the status on this?

File: com.sweethome3d.Desktop.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      </image>
+    </screenshot>
+    <screenshot>
+      <caption>Separate 3D View</caption>
+      <image width="975" height="703">
+        https://raw.githubusercontent.com/hadess/flathub/sweethome3d/sshots/sweethome3d-default-view-virtual-view.png
+      </image>
+    </screenshot>
+    <screenshot>
+      <caption></caption>
```
Comment (bilelmoussaoui) [inline com.sweethome3d.Desktop.appdata.xml]:
```suggestion
```

File: local-build.sh [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+rm -f com.sweethome3d.Desktop.flatpak
+rm -rf _build ; mkdir _build
+rm -rf _repo ; mkdir _repo
```
Comment (bilelmoussaoui) [inline local-build.sh]: Is there a reason for this? once the app is on Flathub the bots are doing a great job so far.

File: com.sweethome3d.Desktop.desktop [hunk 9]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Sweet Home 3D
+Comment=Interior design application
+Type=Application
+Categories=Graphics;3DGraphics;Engineering;
+Exec=/app/bin/SweetHome3D %F
+MimeType=application/x-sweet-home-3d;
+Terminal=false
```
Comment (bilelmoussaoui) [inline com.sweethome3d.Desktop.desktop]: As you intend to upstream everything, I would add some keywords here.

File: com.sweethome3d.Desktop.appdata.xml (hunk 7 above)
Comment (hadess) [reply com.sweethome3d.Desktop.appdata.xml]: I've added a caption instead.

File: local-build.sh (hunk 8 above)
Comment (hadess) [reply local-build.sh]: I have this for every one of the packages I maintain. It stops me from having to remember all the steps. And the bot don't run on my machine...

File: com.sweethome3d.Desktop.desktop (hunk 9 above)
Comment (hadess) [reply com.sweethome3d.Desktop.desktop]: Done.

File: com.sweethome3d.Desktop.appdata.xml (hunk 5 above)
Comment (hadess) [reply com.sweethome3d.Desktop.appdata.xml]: Done.

**Comment by hadess** [comment]
That leaves the thing with the `.Desktop` in the app id. I'm happy putting anything there, but there needs to be some differentiation, as there are more tools and utilities on that same website, so those will need differentiating.

File: com.sweethome3d.Desktop.appdata.xml (hunk 6 above)
Comment (nedrichards) [reply com.sweethome3d.Desktop.appdata.xml]: Should be fine, there's a bunch of special detection for this (telegram is the app you're thinking of)

**Comment by LeandroStanger** [comment]
bot, build com.sweethome3d.Desktop

**Comment by TingPing** [comment]
> That leaves the thing with the .Desktop in the app id. I'm happy putting anything there, but there needs to be some differentiation, as there are more tools and utilities on that same website, so those will need differentiating.

Do those tools have their own names? Its redundant but `com.sweethome3d.Sweethome3d` or such is an accurate id.

**Comment by hadess** [comment]
> Do those tools have their own names? Its redundant but `com.sweethome3d.Sweethome3d` or such is an accurate id.

Done. Would be good if there were clearer guidelines for this.

**Comment by bilelmoussaoui** [comment]
bot, build com.sweethome3d.Sweethome3d

**Comment by LeandroStanger** [comment]
bot, build com.sweethome3d.Sweethome3d

**Comment by TingPing** [comment]
> Done. Would be good if there were clearer guidelines for this.

Added here: http://docs.flatpak.org/en/latest/conventions.html#application-ids

**Comment by hadess** [comment]
> > Done. Would be good if there were clearer guidelines for this.
> 
> Added here: http://docs.flatpak.org/en/latest/conventions.html#application-ids

`.desktop` != `.Desktop` though

> This is a bad ID because the Appstream standard for legacy reasons treats .desktop ending ID’s special causing inconsistenency.

"treat special" isn't correct grammar. Also `IDs` and `inconsistency`.

Let me know if you'd rather I report those problems somewhere else (I don't know where the sources of those files are).

**Comment by TingPing** [comment]
> .desktop != .Desktop though

Yes I do believe this would have worked fine.

> Let me know if you'd rather I report those problems somewhere else (I don't know where the sources of those files are).

https://github.com/flatpak/flatpak-docs

**Comment by hadess** [comment]
> Let me know if you'd rather I report those problems somewhere else (I don't know where the sources of those files are).

https://github.com/flatpak/flatpak-docs/pull/192

**Pull request closed without merging**